syntax = "proto3";

package smartspeech.synthesis.v2;

import "google/protobuf/duration.proto";

option go_package = "./;protocol";

service SmartSpeech {
  rpc Synthesize (stream SynthesisRequest) returns (stream SynthesisResponse);
}

message Options {
  enum AudioEncoding {
    AUDIO_ENCODING_UNSPECIFIED = 0;
    PCM_S16LE = 1;
    OPUS = 2;
    WAV = 3;
    PCM_ALAW = 4;
    G729 = 5;
  }

  AudioEncoding audio_encoding = 1;
  string language = 2;
  string voice = 3;
  bool rebuild_cache = 4;
  bool bypass_cache = 5;
  string fl = 6;
  bool debug_log = 7;
  bool bypass_dl = 8;
}

message Text {
  enum ContentType {
    TEXT = 0;
    SSML = 1;
  }

  string text = 1;
  ContentType content_type = 2;
}

message Audio {
  bytes audio_chunk = 1;
  google.protobuf.Duration audio_duration = 2;
}

message BackendInfo {
  string model_name = 1;
  string model_version = 2;
  string server_version = 3;
}

message SynthesisRequest {
  oneof request {
    Options options = 1;
    Text text = 2;
  }
}

message SynthesisResponse {
  oneof response {
    Audio audio = 1;
    BackendInfo backend_info = 2;
  }
}
